src/routes/publish.ts

```diff
@@ -8,196 +8,602 @@ import { StatsWalesApi } from '../services/stats-wales-api';
+    const lang = req.i18n.language;
+    const statsWalesApi = new StatsWalesApi(lang, req.jwt);
+    const title = req.session.currentTitle;
+    if (!title) {
+        logger.debug('Current title was missing from the session.  Something might have gone wrong');
+        req.session.errors = generateViewErrors(undefined, 400, [generateError('title', 'errors.title_missing', {})]);
+        res.redirect(
+            `/${lang}/${req.i18n.t('routes.publish.start', { lng: lang })}/${t('routes.publish.title', { lng: lang })}`
+        );
+        return;
+    }
+    const file = req.file;
+    if (!file) {
+        generateFileError(req, res);
         return;
     }
 
-    const title: string = req.body.title;
+    const fileName = file.originalname;
+    const fileData = new Blob([file.buffer], { type: file.mimetype });
+    const processedCSV = await statsWalesApi.uploadCSVtoCreateDataset(fileData, fileName, title);
+    await handleProcessedCSV(processedCSV, req, res);
+}
+
+async function uploadNewFileToExistingDataset(req: AuthedRequest, res: Response) {
+    const lang = req.i18n.language;
+    const statsWalesApi = new StatsWalesApi(lang, req.jwt);
+    const currentDataset = checkCurrentDataset(req, res);
+    const currentRevision = checkCurrentRevision(req, res);
+    if (!currentDataset || !currentRevision) {
+        return;
+    }
 
     if (!req.file) {
-        const err: ViewErrDTO = {
-            success: false,
-            status: 400,
-            dataset_id: undefined,
-            errors: [
-                {
-                    field: 'csv',
-                    message: [
-                        {
-                            lang,
-                            message: t('errors.upload.no-csv-data')
-                        }
-                    ],
-                    tag: {
-                        name: 'errors.upload.no-csv-data',
-                        params: {}
-                    }
-                }
-            ]
-        };
-        res.status(400);
-        res.render('publish/upload', err);
+        generateFileError(req, res);
         return;
     }
 
-    const fileName = req.file?.originalname;
-    const fileData = new Blob([req.file?.buffer], { type: req.file?.mimetype });
+    const fileName = req.file.originalname;
+    const fileData = new Blob([req.file.buffer], { type: req.file.mimetype });
+
+    const processedCSV = await statsWalesApi.uploadCSVToFixDataset(
+        currentDataset.id,
+        currentRevision.id,
+        fileData,
+        fileName
+    );
+    await handleProcessedCSV(processedCSV, req, res);
+}
 
-    const processedCSV = await statsWalesApi.uploadCSV(fileData, fileName, title);
+function cleanupSession(req: AuthedRequest) {
+    req.session.currentDataset = undefined;
+    req.session.currentRevision = undefined;
+    req.session.currentImport = undefined;
+    req.session.dimensionCreationRequest = undefined;
+    req.session.errors = undefined;
+    req.session.save();
```

Comment:
I considered that but if we move the users JWT to the session as well for example destroying it might not be a good idea.  Might be worth changing the name as so far this relates specifically to the create journey.  If things go wrong we return to the start, give the errors to the view and then empty the things we need to empty when the user starts a new journey.